common/src/test/java/io/pravega/common/util/PriorityBlockingDrainingQueueTests.java

```diff
@@ -155,7 +155,7 @@ public void testMultiplePriorities() {
         Assert.assertEquals("Unexpected size.", expectedItems.size(), q.size());
         Assert.assertSame("Unexpected peek.", expectedItems.first(), q.peek());
 
-        val finalPoll = q.poll(100);
+        val finalPoll = new ArrayDeque<>(q.poll(100)); // Make a copy since we need to modify it.
```

Comment:
Shouldn't poll remove the item from `q`, in which case it won't matter if you modify it?

Reply:
`poll(100)` returns one of those `SimpleDeque.WrapQueue` instances, to which we cannot add anything. Previously, it returned an `ArrayDeque` to which we could add new items, so the test code was lazily reusing that one. 

Since the `WrapQueue` is closed to additions, I have to make a copy of it. Copying is done using its iterator, so all its elements are contained in the same order. There is no need to test the functionality of `WrapQueue` - that is already done in the `SimpleDequeTests` class.